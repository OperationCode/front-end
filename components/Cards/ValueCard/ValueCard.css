.ValueCard {
  justify-content: flex-start;
  margin: 40px 20px;
  max-width: 425px;
  padding: 2rem;
}

.name {
  text-align: center;
  text-decoration: underline;
}
