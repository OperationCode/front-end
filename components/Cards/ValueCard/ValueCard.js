import React from 'react';
import { string } from 'prop-types';
import Card from 'components/Cards/Card/Card';
import styles from './ValueCard.module.css';

ValueCard.propTypes = {
  name: string.isRequired,
  description: string.isRequired,
};

function ValueCard({ description, name }) {
  return (
    <Card className={styles.ValueCard} hasAnimationOnHover={false}>
      <h3 className={styles.name}>{name}</h3>
      <p>{description}</p>
    </Card>
  );
}

export default ValueCard;
