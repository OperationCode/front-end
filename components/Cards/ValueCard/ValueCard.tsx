import { Card } from 'components/Cards/Card/Card';

export interface ValueCardPropsType {
  name: string;
  description: string;
}

export function ValueCard({ description, name }: ValueCardPropsType) {
  return (
    <Card
      className="!justify-start max-w-[400px] h-[400px] p-8 overflow-y-auto"
      hasAnimationOnHover={false}
    >
      <h3 className="text-center underline">{name}</h3>
      <p>{description}</p>
    </Card>
  );
}
