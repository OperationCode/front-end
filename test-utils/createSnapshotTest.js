/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
