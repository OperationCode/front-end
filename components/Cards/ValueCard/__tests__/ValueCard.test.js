/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ValueCard from '../ValueCard';

describe('ValueCard', () => {
  test('should render properly with all required props assigned', () => {
    createSnapshotTest(
      <ValueCard
        name="Testing"
        description="Testing is good for the soul and scientifically proven to make puppies happy."
      />,
    );
  });
});
