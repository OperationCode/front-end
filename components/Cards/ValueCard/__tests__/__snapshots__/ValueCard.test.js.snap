// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueCard should render properly with all required props assigned 1`] = `
<article
  className="Card ValueCard"
>
  <h3
    className="name"
  >
    Testing
  </h3>
  <p>
    Testing is good for the soul and scientifically proven to make puppies happy.
  </p>
</article>
`;
